### PR TITLE
NL structural print

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/2560.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/UtilityDataStructures/LazyList.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace Microsoft.PowerFx.Core.UtilityDataStructures
     /// the operation without the creation of many intermediate large
     /// memory blocks.
     /// </summary>
-    internal class LazyList<T> : IEnumerable<T>
+    public class LazyList<T> : IEnumerable<T>
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/UtilityDataStructures/LazyList.cs:17]: >public [](http://example.com/codeflow?start=4&length=6)

I'm skeptical of the value of this class... wondering if we can avoid making it public.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs [hunk 2]
```diff
@@ -672,6 +670,18 @@ public string ApplyGetLogging()
             return _expressionAnonymous;
         }
 
+        internal string ApplyGetLogging(ISanitizedNameProvider nameProvider = null)
+        {
+            if (_expressionAnonymous == null)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs:675]: >_expressionAnonymous [](http://example.com/codeflow?start=16&length=20)

We shouldn't reuse _expressionAnonymous cache here. 

Else calling ApplyGetLogging(sanitizer) will change the result of ApplyGetLogging(). 

or even calling: 
ApplyGetLogging(santizer1) 
ApplyGetLogging(santizer2)  // will fail and return sanitizer1 result. 

Probably fine to just not cache here.

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs:675]: Can you add a test that illustrates this problem? That will help verify the fix.  :)

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs [hunk 3]
```diff
@@ -672,6 +670,12 @@ public string ApplyGetLogging()
             return _expressionAnonymous;
         }
 
+        internal string ApplyGetLogging(ISanitizedNameProvider nameProvider = null)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs:673]: Which overload does just calling `ApplyGetLogging()` bind against?  Line 661, or line 673 with nameProvider=null?

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs:673]: Having 2 separate methods is import from a breaking change perspective, but the trip up is the default value for the param.  Basically:
- should we remove the default value for the param?
- should the existing call in ApplyGetLogging() just become a call to ApplyGetLogging(null).

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**